### PR TITLE
For issue #5298 the same page appears in the History section multiple times

### PR DESCRIPTION
**Pull request by Amejia481**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/test/java/org/mozilla/fenix/components/history/PagedHistoryProviderTest.kt [hunk 1]
```diff
@@ -452,4 +452,62 @@ class PagedHistoryProviderTest {
...
+                title = "No duplicate item 2",
+                url = "url",
+                visitedAt = 0
+            ),
+            History.Group(
+                id = 5,
+                title = "Group 5",
+                visitedAt = 0,
+                items = emptyList()
+            ),
```
Comment (csadilek) [inline app/src/test/java/org/mozilla/fenix/components/history/PagedHistoryProviderTest.kt:488]: nit: Let's add another (single) item with the same url at the end to test that we really only filter consecutive duplicates.

File: app/src/test/java/org/mozilla/fenix/components/history/PagedHistoryProviderTest.kt [hunk 2]
```diff
@@ -452,4 +452,62 @@ class PagedHistoryProviderTest {
...
+                title = "No duplicate item",
+                url = "url",
+                visitedAt = 0
+            ),
+            History.Group(
+                id = 5,
+                title = "Group 5",
+                visitedAt = 0,
+                items = emptyList()
+            ),
```
Comment (csadilek) [inline app/src/test/java/org/mozilla/fenix/components/history/PagedHistoryProviderTest.kt:509]: And here ^^ as well to verify it's still in the result after.

**Review by csadilek — APPROVED** [review]
👍 

One suggestion below to improve the test. Overall, I think this makes the history view better and we can live with the edge case of still seeing duplicates when there's not enough history to populated a full page yet.

**Comment by Amejia481** [comment]
> 👍
> 
> One suggestion below to improve the test. Overall, I think this makes the history view better and we can live with the edge case of still seeing duplicates when there's not enough history to populated a full page yet.

👍🏽

File: app/src/test/java/org/mozilla/fenix/components/history/PagedHistoryProviderTest.kt (hunk 2 above)
Comment (Amejia481) [reply app/src/test/java/org/mozilla/fenix/components/history/PagedHistoryProviderTest.kt:515]: Added :)!

File: app/src/test/java/org/mozilla/fenix/components/history/PagedHistoryProviderTest.kt (hunk 1 above)
Comment (Amejia481) [reply app/src/test/java/org/mozilla/fenix/components/history/PagedHistoryProviderTest.kt:488]: Added :)!

**Review by csadilek — APPROVED** [review]
